### PR TITLE
(feat) Add debouncing to patient list search and fix search component height on tablet 

### PR DESCRIPTION
**Pull request by Jexsie**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
- Adds debouncing to list searching
- Fixes the height of the search component inside the patient tablet from `xl` to `lg` on tablet

## Screenshots

*None.*

Screenshot from 2023-08-07 11-45-08

image

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by Jexsie** [comment]
cc @vasharma05

File: packages/esm-patient-list-app/src/patient-list-list/patient-list-table.component.tsx [hunk 1]
```diff
@@ -85,10 +84,14 @@ const PatientListTableContainer: React.FC<PatientListTableContainerProps> = ({
...
-    setSearchTerm(str);
-    goTo(1);
-  };
+  const handleSearch = useMemo(
+    () =>
+      debounce((searchTerm) => {
+        goTo(1);
+        setSearchTerm(searchTerm);
+      }, 300),
+    [],
```
Comment (vasharma05) [inline packages/esm-patient-list-app/src/patient-list-list/patient-list-table.component.tsx:93]: goTo and searchTerms are a dependency

**Review by vasharma05 — COMMENTED** [review]
LGTM!
Just a small change

File: packages/esm-patient-list-app/src/patient-list-list/patient-list-table.component.tsx (hunk 1 above)
Comment (Jexsie) [reply packages/esm-patient-list-app/src/patient-list-list/patient-list-table.component.tsx]: Thanks

**Comment by denniskigen** [comment]
@vasharma05, can this leverage the `useDebounce` hook you added to Core?

**Comment by Jexsie** [comment]
i think i can handle it @vasharma05 . Please alsi take a look at my other PRs

**Comment by denniskigen** [comment]
Closing as this is obviated by https://github.com/openmrs/openmrs-esm-patient-management/pull/865.

**Pull request closed without merging**